service/frontend/handler.go

```diff
@@ -391,6 +402,9 @@ func (wh *WorkflowHandler) RecordActivityTaskHeartbeat(
 	sw := wh.startRequestProfile(scope)
 	defer sw.Stop()
 
+	// Count the request in the RPS, but we still accept it even if RPS is exceeded
```

Comment:
Isn't it safer to throttle? Is there a downside to throttling uniformly across APIs?
I understand the way cadence is supposed to be used, these APIs should not be heavily called but we will often not have control over user code and they can mistakenly or with wrong usage be able to take down the service.